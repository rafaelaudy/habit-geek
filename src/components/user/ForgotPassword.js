import React, { useState, Fragment } from "react";
import { useTranslation } from "react-i18next";
import * as userService from "@rafael.audy/habit-geek-utils/services/userService";

const ForgotPassword = () => {
  const [email, setEmail] = useState("");
  const [message, setMessage] = useState("");
  const [validityChecked, setValidityChecked] = useState(false);
  const { t } = useTranslation();

  const forgotPassword = event => {
    event.preventDefault();
    setValidityChecked(true);

    if (event.target.checkValidity()) {
      userService
        .forgotPassword(email)
        .then(() => setMessage(t("forgot-password-email-sent-message")))
        .catch(() => setMessage(t("forgot-password-email-not-sent")));
    }
  };

  return (
    <div className="mobile-size-container">
      <form
        className={`${validityChecked ? "was-validated" : ""}`}
        onSubmit={forgotPassword}
        noValidate
      >
        <h2>{t("forgot-password-title")}</h2>

        {message ? (
          <Fragment>
            <div>{message}</div>

            {t("forgot-password-try-again-label")}
            <button
              onClick={() => setMessage(undefined)}
              className="px-0 btn btn-link"
            >
              {t("forgot-password-try-again")}
            </button>
          </Fragment>
        ) : (
          <div className="mb-3">
            <label htmlFor="user-form-email">{t("user-form-email")}</label>
            <input
              id="user-form-email"
              className="form-control"
              value={email}
              onChange={e => setEmail(e.target.value)}
              required
            ></input>
            <div className="invalid-feedback">
              {t("user-form-email-validation")}
            </div>
            <hr className="mb-4" />
            <button type="submit" className="btn btn-primary btn-lg btn-block">
              {t("forgot-password-forgot-password")}
            </button>
          </div>
        )}
      </form>
    </div>
  );
};

export default ForgotPassword;
