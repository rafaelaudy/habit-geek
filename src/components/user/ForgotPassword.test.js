import React from "react";
import { shallow } from "enzyme";
import ForgotPassword from "./ForgotPassword";

//TODO
describe("ForgotPassword component", () => {
  it("Renders static elements", () => {
    const forgotPassword = shallow(<ForgotPassword />);
    expect(forgotPassword).toMatchSnapshot();
  });
});
