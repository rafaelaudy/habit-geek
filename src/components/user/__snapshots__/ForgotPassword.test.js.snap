// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ForgotPassword component Renders static elements 1`] = `
<div
  className="mobile-size-container"
>
  <form
    className=""
    noValidate={true}
    onSubmit={[Function]}
  >
    <h2>
      forgot-password-title
    </h2>
    <div
      className="mb-3"
    >
      <label
        htmlFor="user-form-email"
      >
        user-form-email
      </label>
      <input
        className="form-control"
        id="user-form-email"
        onChange={[Function]}
        required={true}
        value=""
      />
      <div
        className="invalid-feedback"
      >
        user-form-email-validation
      </div>
      <hr
        className="mb-4"
      />
      <button
        className="btn btn-primary btn-lg btn-block"
        type="submit"
      >
        forgot-password-forgot-password
      </button>
    </div>
  </form>
</div>
`;
